src/kvp/mod.rs

```diff
@@ -337,6 +348,85 @@ where
         self.0.insert(kvp);
         self
     }
+
+    pub fn contains(&self, kvp: KeyValuePair<V>) -> bool {
+        self.0.contains(&kvp)
+    }
+
+    pub fn contains_raw(
+        &self,
+        key: impl AsRef<str>,
+        value: impl AsRef<str>,
+    ) -> Result<bool, KeyValuePairsError<V::Error>> {
+        let kvp = KeyValuePair::try_from((key.as_ref(), value.as_ref()))
+            .context(KeyValuePairParseSnafu)?;
+
+        Ok(self.0.contains(&kvp))
+    }
+
+    pub fn contains_all(&self, kvps: KeyValuePairs<V>) -> bool {
+        for kvp in kvps {
+            if !self.contains(&kvp) {
+                return false;
+            }
+        }
+
+        true
+    }
+
+    pub fn contains_all_raw(
+        &self,
+        keys: impl AsRef<[str]>,
+        values: impl AsRef<[str]>,
+    ) -> Result<bool, KeyValuePairsError<V::Error>> {
+        for (key, value) in keys.as_ref().iter().zip(values.as_ref()) {
+            if !self
+                .contains_raw(key, value)
+                .context(KeyValuePairParseSnafu)?
+            {
+                return Ok(false);
+            }
+        }
+
+        Ok(true)
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct ObjectLabels<'a, T> {
```

Comment:
nit: Add a doc comment for what this is used for. Do they represent the existing labels on the resource?

Reply:
Yes, this is part of the above todo list.

Reply:
Added in 1e6979f.